### DISCUSSION TITLE
Directed graph options

### DISCUSSION THREAD

**Discussion by simonmesmith** [reflex-dev/reflex#1245] [General]
I'd like to add a directed graph (e.g. [NetworkX](https://networkx.org/), [Cytoscape](https://cytoscape.org/)) to a page, but am struggling to get this to work.

I tried implementing this with NetworkX and then rendering it via Plotly, but could not get it to be properly interactive. Ideally I'd like access to things like the on_click event for nodes.

I also tried wrapping the [react-cytoscapejs](https://www.npmjs.com/package/react-cytoscapejs) component, but I received a "SyntaxError: Cannot use import statement outside a module" issue that I couldn't resolve despite trying several fixes.

Has anyone implemented a directed graph?

**Comment by simonmesmith** [comment]
Update: Also tried [react-force-graph](https://www.npmjs.com/package/react-force-graph) and encountered errors that I could not resolve.

**Comment by Alek99** [comment]
Got it thanks for the suggestion, I'm currently working on scoping how to make our graphing better which libraries to support and types of graphs.
